vagrant/roles/local.defaults/vars/main.yml

```diff
@@ -134,3 +135,33 @@ environments:
         networks:
           public: 5
         groups: [clients]
+
+  cephfs:
+    cpus: 2
+    memory: 1024
+
+    data:
+      release: quincy
```

Comment:
Are there packages for `main` in the centos storage sig builds? It looks to me that the only place the variable is used. AFAICT there may not be one for reef yet either...

There are other ways to get ceph installed, but I'd be happy merging something first and then incrementally improving them, so I'd be happy enough to start with main or reef or even qunicy and then discussing what our defaults should be and what options we should provide.

Reply:
We've managed to configure Ceph latest(main) RPM repository in [samba-build](https://github.com/samba-in-kubernetes/samba-build/blob/main/ansible/roles/prep.dirs/tasks/repo.yml) repository. We could then make use of the images from _quay.ceph.io/ceph-ci/ceph:main_ for `cephadm` installed from such a repository.

Reply:
OK, I wasn't aware of that for samba-build. I generally don't install any ceph packages on my test clusters and so I don't think it's strictly needed but I won't object if you all think it's simpler.

Reply:
Ok. I'll try to use the images from quay.

Reply:
If you want to install on fedora you may want to consider a "packageless" install then. This can be done by fetching the cephadm binary from the el9 build (the el9-ness doesn't matter) and just running everything else using containers. You could also get the fedora ceph packages but then you won't be testing with the latest `main` components.  

A packageless install will work on all platforms AFAIK. You just need to replace any direct calls to `ceph` command line tool with `cephadm shell -- ceph`.  Let me know if you want more info. This is how I deploy ceph for my local testing.

Reply:
I've gone the cephadm path, which installs all ceph components from main branch. However the samba server runs on the host and still requires ceph-common to be present so that the mount can be done. I discussed this with @phlogistonjohn and we believe that user-mode ceph components are only used for argument parsing and initiate the mount. Once completed, only the kernel module should be active, so version mismatch between host's ceph packages and the ceph containers installed shouldn't be a problem.

In any case I've updated the release to 'reef'.

As an alternative we could try to see if it's feasible to mount the ceph volume inside a container and make it visible to the host. This way the host doesn't require any user-mode ceph components. In any case this will be left for another PR if we decide to do so.